Classes/Fluid/ViewHelper/ComponentRenderer.php

```diff
@@ -221,7 +261,12 @@ function () use ($componentFile) {
         }
 
         // Render component
-        return $this->parsedTemplate->render($renderingContext);
+        $renderedComponent = $this->parsedTemplate->render($renderingContext);
+
+        // Restore original component context
+        $viewHelperVariableContainer->add(static::class, 'componentContext', $parentComponentContext);
```

Comment:
Maybe this can be optimized so that the merging and copying is only necessary in case context is actually in use?